API-Samples/init_texture/init_texture.cpp

```diff
@@ -97,9 +97,12 @@ int sample_main(int argc, char *argv[]) {
     image_create_info.mipLevels = 1;
     image_create_info.arrayLayers = 1;
     image_create_info.samples = NUM_SAMPLES;
-    image_create_info.tiling = VK_IMAGE_TILING_LINEAR;
+    image_create_info.tiling = texObj.needs_staging ? VK_IMAGE_TILING_OPTIMAL : VK_IMAGE_TILING_LINEAR;
     image_create_info.initialLayout = VK_IMAGE_LAYOUT_PREINITIALIZED;
```

Comment:
Is in VK_IMAGE_LAYOUT_PREINITIALIZED even if we are copying to it with a staging buffer?